storage/innobase/handler/handler0alter.cc

```diff
@@ -7978,6 +7854,21 @@ ha_innobase::prepare_inplace_alter_table(
 		}
 	}
 
+	if (ha_alter_info->handler_flags & ALTER_RENAME_INDEX) {
+		for (const Alter_inplace_info::Rename_key_pair& pair :
+		     ha_alter_info->rename_keys) {
+			dict_index_t* index = dict_table_get_index_on_name(
+			    indexed_table, pair.old_key->name.str);
+			ut_ad(index);
+
+			if (index->is_corrupted()) {
+				my_error(ER_INDEX_CORRUPT, MYF(0),
+					 index->name());
+				goto err_exit;
+			}
```

Comment:
I think that we should be more fault-tolerant here:
```c++
if (!index || index->is_corrupted()) {
	my_error(ER_INDEX_CORRUPT, MYF(0), pair.old_key->name.str);
	goto err_exit;
}
```

Reply:
This one is fixed.